### DISCUSSION TITLE
JavaScript over TypeScript for SvelteKit codebase

### DISCUSSION THREAD

**Discussion by hanneswidrig** [sveltejs/kit#4429] [Q&A] [answered]
Hello, not sure if this is the correct spot to ask this question. I am curious why SvelteKit appears to written in JavaScript while Svelte itself is written in TypeScript?

**Comment by geoffrich** [comment, marked as answer]
Note that SvelteKit is still typed, but uses JSDoc instead of TypeScript. You can find some tweets from Rich Harris about this, but it comes down to a few reasons:

- You can still generate types from JSDoc source code and the resulting code is generally smaller than TS-compiled code ([source](https://twitter.com/Rich_Harris/status/1350436286948122625))
- Using JS made it easier to make changes when the project was in an experimental phase ([source](https://twitter.com/Rich_Harris/status/1487933546966753284))
- Using JSDoc types means you can run tests without adding a build step ([source](https://twitter.com/Rich_Harris/status/1440447529934417923))

Maintainers have [said in the past](https://twitter.com/antony/status/1350436771134369795) that Kit could go back to TypeScript after 1.0, but so far JSDoc types are working well for the project.
